File: 0090-subsets-ii/0090-subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        subsets = [[]]; d = {}
        for num in nums:
            for i in range(len(subsets)):
                currSubset = subsets[i]
                subsets.append(currSubset + [num])
        for sub in subsets: 
            sub.sort();
            d[tuple(sub)] = 1;
        subsets.clear();
        for sub in d.keys(): subsets.append(list(sub));
        return subsets

File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
    vector<vector<int>> subsetsWithDup(vector<int>& arr) {
        vector<vector<int>> fin;
        map<vector<int>, long int> mp;
        long long int i, j, n = arr.size();
        for (i = 0; i < pow(2, n); i++)
        {
            vector<int> b;
            for (j = 0; j < n; j++)
            {
                if (i & (1<<j))
                {
                    b.push_back(arr[j]);
                }
            }
            sort(b.begin(), b.end());
            mp[b] ++;
        }
        for (auto it: mp)
        {
            fin.push_back(it.first);
        }
        return fin;
    }
};